nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/ConfigParser.java

```diff
@@ -161,36 +182,55 @@ public DomDocument parse(URL source, DomDocument document, Dom parent) {
      *      Null if the XML element didn't yield anything (which can happen if the element is skipped.)
      *      Otherwise fully parsed valid {@link Dom} object.
      */
-    protected Dom handleElement(XMLStreamReader in,DomDocument document, Dom parent) throws XMLStreamException {
+    protected Dom handleElement(XMLStreamReader in, DomDocument document, Dom parent) throws XMLStreamException {
         ConfigModel model = document.getModelByElementName(in.getLocalName());
         if(model==null) {
+            // Get the element name
             String localName = in.getLocalName();
-            Logger.getAnonymousLogger().severe("Ignoring unrecognized element "+in.getLocalName() + " at " + in.getLocation());
+
+            log(SEVERE, "Ignoring unrecognized element %s at %s", in.getLocalName(), in.getLocation());
+
             // flush the sub element content from the parser
             int depth=1;
             while(depth>0) {
                 final int tag = in.nextTag();
                 if (tag==START_ELEMENT && in.getLocalName().equals(localName)) {
-                    if (Logger.getAnonymousLogger().isLoggable(Level.FINE)) {
-                        Logger.getAnonymousLogger().fine("Found child of same type "+localName+" ignoring too");
-                    }
+                    log(FINE, "Found child of same type %s ignoring too", localName);
                     depth++;
                 }
                 if (tag==END_ELEMENT && in.getLocalName().equals(localName)) {
-                    if (Logger.getAnonymousLogger().isLoggable(Level.FINE)) {
-                        Logger.getAnonymousLogger().fine("closing element type " + localName);
-                    }
+                    log(FINE, "Closing element type %s", localName);
                     depth--;
                 }
-                if (Logger.getAnonymousLogger().isLoggable(Level.FINE) && tag==START_ELEMENT) {
-                    Logger.getAnonymousLogger().fine("Jumping over " + in.getLocalName());
+                if (tag == START_ELEMENT) {
+                    log(FINE, "Jumping over %s", in.getLocalName());
```

Comment:
LABEL 1: logging fine message as an error, because log method will add this message to errors if it is loggable.